### PR TITLE
Support OpExtension in frontends

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Support OpExtension in frontends*
 - *Support OPENVINO_FRAMEWORK_MAP for PyTorch FE*

### Tickets:
 - *CVS-132471*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]
should we cover by tests the original issue with C++ macro `OPENVINO_FRAMEWORK_MAP(pytorch)` ?

**Comment by mvafin** [comment]
@ilya-lavrenov Agree, will add test

**Review by eaidova — APPROVED** [review]

**Review by praasz — APPROVED** [review]

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py [hunk 1]
```diff
@@ -112,11 +120,69 @@ def create_ref_graph2():
...
+    @pytest.mark.parametrize("params", test_data)
+    @pytest.mark.nightly
+    @pytest.mark.precommit
+    def test_pt_mo_convert_extensions(self, params, ie_device, precision, ir_version,
+                                   temp_dir, use_legacy_frontend):
+        model = self.create_model(temp_dir)
+
+        test_params = params['params_test']
+        test_params.update({'input_model': model})
+        self._test_by_ref_graph(temp_dir, test_params, params['ref_graph'])
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py:188]: Please add the same test for TF Frontend

File: src/frontends/tensorflow/src/frontend.cpp [hunk 2]
```diff
@@ -559,5 +559,9 @@ void FrontEnd::add_extension(const std::shared_ptr<ov::Extension>& extension) {
                    std::dynamic_pointer_cast<ov::frontend::tensorflow::ConversionExtension>(extension)) {
         m_conversion_extensions.push_back(tensorflow_conv_ext);
         m_op_translators[tensorflow_conv_ext->get_op_type()] = tensorflow_conv_ext->get_converter();
+    } else if (auto op_base_ext = std::dynamic_pointer_cast<ov::BaseOpExtension>(extension)) {
+        for (const auto& attached_ext : op_base_ext->get_attached_extensions()) {
+            add_extension(attached_ext);
```
Comment (rkazants) [inline src/frontends/tensorflow/src/frontend.cpp:564]: Why just we update Op extension? I think we should also update `m_op_translators` dictionary.

**Review by rkazants — CHANGES_REQUESTED** [review]
Not sure that it works for TF FE

File: src/frontends/tensorflow/src/frontend.cpp (hunk 2 above)
Comment (mvafin) [reply src/frontends/tensorflow/src/frontend.cpp:564]: I don't think so, we extract `ConversionExtension` from `OpExtension` and add it using `add_extension` so we do not need to do anything extra.

**Review by PiotrKrzem — APPROVED** [review]

File: src/frontends/pytorch/src/frontend.cpp [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 #include "input_model.hpp"
 #include "op_table.hpp"
+#include "openvino/core/op_extension.hpp"
```
Comment (mvafin) [inline src/frontends/pytorch/src/frontend.cpp:9]:
```suggestion
```

**Review by rkazants — APPROVED** [review]

**Pull request merged**